packages/react/src/components/Box/Box.tsx

```diff
@@ -1,49 +1,70 @@
 import * as React from 'react'
-import {
-  childrenExist,
-  createShorthandFactory,
-  UIComponentProps,
-  ContentComponentProps,
-  ChildrenComponentProps,
-  commonPropTypes,
-  rtlTextContainer,
-} from '../../lib'
-import createComponent, { CreateComponentReturnType } from '../../lib/createComponent'
-import { ReactProps } from '../../types'
-
-export interface BoxProps
-  extends UIComponentProps<BoxProps>,
-    ContentComponentProps,
-    ChildrenComponentProps {}
+import { createShorthandFactory, callable, getUnhandledProps } from '../../lib'
+
+import { BoxClassName, BoxProps, boxPropTypes, renderBox } from './common'
+import { FelaTheme } from 'react-fela'
+import { IRenderer as FelaRenderer } from 'fela'
+import cx from 'classnames'
+
+const withThemeContext = render => <FelaTheme>{(theme: any) => render(theme)}</FelaTheme>
+
+type BoxRenderConfig = {
+  ElementType: React.ReactType
+  props: BoxProps
+  renderer: FelaRenderer
+}
+
+const render = ({ ElementType, props, renderer }: BoxRenderConfig) => {
+  const { styles, className: predefinedClasses, content, children } = props
+
+  const unhandledProps = getUnhandledProps({ handledProps }, props)
+  const classes = {
+    root: renderer.renderRule(callable(styles), props),
```

Comment:
I guess the main optimization is somewhere here, can you explain in a few words how it's achieved?

Reply:
note that here we imply that styles is an already prepared object, so that we don't need to
- consume theme styles, merge them
- merge theme variables
- resolve styles with merged variables

The thing we are requiring by optimised box is set of already resolved styles - and this is, essentially, the thing that we always have in Stardust code